shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -71,13 +83,40 @@ public void clearClient() {
                 clearTextInputClient();
             }
         });
+        this.platformViewsController = platformViewsController;
+        platformViewsController.attachTextInputPlugin(this);
```

Comment:
Do we have the constructor vs attach behavior backwards?

Right now we're saying a `TextInputPlugin` should always have a `PlatformViewsController`, i.e., the plugin makes no sense without a `PlatformViewsController`. But we're saying a `PlatformViewsController` may or may not have an associated `TextInputPlugin`, e.g., attach/detach. I just wanted to confirm that such a statement is intended to be accurate?

The alternative would be that a `PlatformViewsController` always has a `TextInputPlugin`, but a `TextInputPlugin` may or may not have a `PlatformViewsController`.

Or the third option is that they vary independently and both of the classes offer `attach()` methods for the other.

Is the relationship in this PR right now definitely the one you want?

Reply:
Yeah that relationship is correct.
A platform views controller can live without a TextInputPlugin (e.g we can maintain a WebView instance as we're detached from an Activity and use it later when attached).

The TextInputPlugin cannot function properly without a reference to a PlatformViewsController(as it might get a request to set a platform view client).

Reply:
Sounds good.